packages/webapp/src/containers/Finances/useCropSaleInputs.jsx

```diff
@@ -0,0 +1,322 @@
+import CropSaleItem from '../../components/Forms/GeneralRevenue/CropSaleItem';
+import { STATUS } from '../../components/Forms/GeneralRevenue/constants';
+import { useTranslation } from 'react-i18next';
+import styles from '../../components/Forms/GeneralRevenue/styles.module.scss';
+import { useSelector } from 'react-redux';
+import { currentAndPlannedManagementPlansSelector } from '../managementPlanSlice';
+import { measurementSelector } from '../userFarmSlice';
+
+/**
+ * Reformat the sale data for ease of use with react-hook-forms.
+ *
+ * @param {Object} sale - The sale data object to be reformatted.
+ * @returns {Object} An object where crop variety IDs are keys, and values contain properties:
+ * - crop_variety_id: The unique identifier for the crop variety.
+ * - quantity: The quantity of the crop variety sold.
+ * - quantity_unit: The unit of measurement for the sold quantity.
+ * - sale_value: The value of the crop variety sold.
+ */
+const reformatSaleData = (sale) => {
+  // Reformat selector to match component format
+  // TODO: match component to selector format
+  return sale?.crop_variety_sale?.reduce(
+    (acc, cur) => ({
+      ...acc,
+      [cur.crop_variety_id]: {
+        crop_variety_id: cur.crop_variety_id,
+        quantity: cur.quantity,
+        quantity_unit: cur.quantity_unit,
+        sale_value: cur.sale_value,
+      },
+    }),
+    {},
+  );
+};
+
+/**
+ * Get filter options for crop varieties based on provided management plans and translations.
+ *
+ * This function generates an array of crop variety filter options using the provided
+ * management plans and translation function. It ensures that the options are unique
+ * based on crop variety IDs and sorts them alphabetically by label.
+ *
+ * @param {Object[]} managementPlans - An array of management plans containing crop variety data.
+ * @param {Function} t - The translation function used to translate crop variety names.
+ * @returns {Object[]} An array of crop variety filter options with unique crop variety IDs and translated labels.
+ * Each option object contains:
+ * - label: The translated label for the crop variety, formatted as "Crop Variety Name, Translated Crop Name".
+ * - value: The unique identifier for the crop variety.
+ */
+const getCropVarietyFilterOptions = (managementPlans, t) => {
+  if (!managementPlans || managementPlans.length === 0) {
+    return [];
+  }
+
+  let cropVarietyFilterOptions = [];
+  let cropVarietySet = new Set();
+
+  for (let mp of managementPlans) {
+    if (!cropVarietySet.has(mp.crop_variety_id)) {
+      cropVarietyFilterOptions.push({
+        label: mp.crop_variety_name
+          ? `${mp.crop_variety_name}, ${t(`crop:${mp.crop_translation_key}`)}`
+          : t(`crop:${mp.crop_translation_key}`),
+        value: mp.crop_variety_id,
+      });
+      cropVarietySet.add(mp.crop_variety_id);
+    }
+  }
+
+  cropVarietyFilterOptions.sort((a, b) => (a.label > b.label ? 1 : b.label > a.label ? -1 : 0));
```

Comment:
Would recommend good old if/elses here to make this a bit easier to read

Reply:
Thats a copy paste special for sure -- I will try haha

Reply:
hopefully fixed